zds/member/tests/tests_views.py

```diff
@@ -1805,19 +1796,26 @@ def test_same_ipv6_network_but_different_ip(self) -> None:
     def test_different_ipv6_network(self) -> None:
         self.client.force_login(self.staff)
         response = self.client.get(reverse(member_from_ip, args=[self.user_ipv6_different_network.last_ip_address]))
-        self.assertEqual(response.status_code, 200)
-        self.assertTemplateUsed(response, "member/admin/memberip.html")
         self.assertContains(response, self.user_ipv6_different_network.user.username)
         self.assertNotContains(response, self.user_ipv6_same_ip_1.user.username)
         self.assertNotContains(response, self.user_ipv6_same_ip_2.user.username)
         self.assertNotContains(response, self.user_ipv6_same_network.user.username)
 
-    def test_access_rights_to_ip_page(self) -> None:
-        regular_user = ProfileFactory()
-        regular_user.user.save()
-        self.client.force_login(regular_user.user)
-        reponse = self.client.get(reverse(member_from_ip, args=["0.0.0.0"]))
-        self.assertEqual(reponse.status_code, 403)
-        self.client.logout()
-        reponse = self.client.get(reverse(member_from_ip, args=["0.0.0.0"]))
-        self.assertEqual(reponse.status_code, 302)
+    def test_access_rights_to_ip_page_as_regular_user(self) -> None:
+        self.client.force_login(self.regular_user.user)
+        response = self.client.get(reverse(member_from_ip, args=["0.0.0.0"]))
+        self.assertEqual(response.status_code, 403)
+
+    def test_access_rights_to_ip_page_as_non_user(self) -> None:
```

Comment:
Petit détail, mais sous Django, la terminologie usuelle pour “_non user_” c'est _anonymous_. Rester cohérent avec les conventions permet de garder un code plus facile à lire pour celles et ceux qui passeront derrière.

```suggestion
    def test_access_rights_to_ip_page_as_anonymous(self) -> None:
```